test/trafos/convolution_test.py

```diff
@@ -0,0 +1,292 @@
+    true_range = odl.uniform_discr_fromdiscr(discr, min_corner=[-0.4, 0.15])
+    assert conv_small.range.partition.approx_equals(true_range.partition,
+                                                    atol=1e-6)
+
+
+def test_real_conv_kernel():
+
+    discr = odl.uniform_discr([-1, 0], [1, 1], [5, 10])
+    subdiscr = odl.uniform_discr_fromdiscr(discr, min_corner=[0, 0],
+                                           nsamples=[3, 3])
+    std_ker_discr = odl.uniform_discr([-1, -0.5], [1, 0.5], [5, 10])
+
+    full_ker = discr.one()
+    full_ker_arr = full_ker.asarray()
+    small_ker = subdiscr.one()
+    small_ker_arr = small_ker.asarray()
+
+    def func_ker(x):
+        return sum(x)
+
+    conv_full = odl.trafos.RealSpaceConvolution(discr, full_ker)
+    assert conv_full.kernel() == full_ker
+
+    conv_full_arr = odl.trafos.RealSpaceConvolution(discr, full_ker_arr)
+    assert np.array_equal(conv_full_arr.kernel(), full_ker_arr)
+    assert np.allclose(conv_full_arr.kernel().space.partition.midpoint, 0)
+    assert conv_full_arr.kernel().shape == discr.shape
+
+    conv_small = odl.trafos.RealSpaceConvolution(discr, small_ker)
+    assert conv_small.kernel() == small_ker
+
+    conv_small_arr = odl.trafos.RealSpaceConvolution(discr, small_ker_arr)
+    assert np.array_equal(conv_small_arr.kernel(), small_ker_arr)
+    assert np.allclose(conv_small_arr.kernel().space.partition.midpoint, 0)
+    assert conv_small_arr.kernel().shape == subdiscr.shape
+
+    conv_func = odl.trafos.RealSpaceConvolution(discr, func_ker)
+    assert conv_func.kernel().space == std_ker_discr
+    assert all_almost_equal(conv_func.kernel(),
+                            std_ker_discr.element(func_ker))
+
+    def func_ker_p(x, **kwargs):
+        p = kwargs.pop('p', 1.0)
+        return x[0] + p * x[1]
+
+    conv_func_p = odl.trafos.RealSpaceConvolution(discr, func_ker_p)
+    assert all_almost_equal(conv_func_p.kernel(),
+                            std_ker_discr.element(func_ker_p, p=1.0))
+
+    conv_func_p = odl.trafos.RealSpaceConvolution(
+        discr, func_ker_p, kernel_kwargs={'p': 0.5})
+    assert all_almost_equal(conv_func_p.kernel(),
+                            std_ker_discr.element(func_ker_p, p=0.5))
+
+
+def test_real_conv_resampling():
+
+    discr = odl.uniform_discr([-1, 0], [1, 1], [5, 10])
+    ker_discr = odl.uniform_discr([-1, -0.25], [1, 0.25], [20, 2])
+    kernel = ker_discr.one()
+
+    cs_up = [0.1, 0.1]
+    discr_up = odl.uniform_discr_fromdiscr(discr, cell_sides=cs_up)
+    ker_discr_up = odl.uniform_discr_fromdiscr(ker_discr, cell_sides=cs_up)
+    conv_up = odl.trafos.RealSpaceConvolution(discr, kernel, resample='up')
+    assert conv_up._domain_resampling_op.domain == discr
+    assert conv_up._domain_resampling_op.range == discr_up
+    assert conv_up._kernel_resampling_op.domain == ker_discr
+    assert conv_up._kernel_resampling_op.range == ker_discr_up
+
+    cs_down = [0.4, 0.25]
```

Comment:
Frankly I have no idea if this is correct or not. Could you try adding a few lines that tell us why?